.github/workflows/changelog-check.yml

```diff
@@ -21,10 +21,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
       - name: Check for changelog entry in diff
         run: |
+          # by default the checkout action doesn't checkout all branches
+          git fetch origin ${{ github.event.pull_request.base.ref }}
```

Comment:
This doesn't do a checkout, right? So the diff will never report any changes in files because it's diffing the same commit?

I think there needs to be a `git checkout` after this, right? Or you need to diff against `FETCH_HEAD` ?

Reply:
I'm looking into it on a test repo, something is still not quite right with the ref checkout... it's checking out master and not the PR.

 I tried it with a fetch and comparing between `origin/<base.ref>` and that worked though so this should be OK without a checkout.

Reply:
Oh I see, that's expected 🤦 https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target the `GITHUB_REF` for pull_request_target (base ref) is not the same as pull_request (merge commit).